src/Interpreters/Streaming/HashJoin.cpp

```diff
@@ -1027,77 +1055,71 @@ void HashJoin::postInit(const Block & left_header, const Block & output_header_,
-                {
-                    if (col.name == ProtonConsts::RESERVED_DELTA_FLAG)
-                        left_delta_column_position_rlj = i;
-                    else if (col.name.ends_with(ProtonConsts::RESERVED_DELTA_FLAG))
-                        right_delta_column_position_rlj = i;
+            for (size_t i = 0; const auto & col : right_join_left_header)
+            {
+                if (col.name == ProtonConsts::RESERVED_DELTA_FLAG)
+                    left_delta_column_position_rlj = i;
+                else if (col.name.ends_with(ProtonConsts::RESERVED_DELTA_FLAG))
+                    right_delta_column_position_rlj = i;
 
-                    ++i;
-                }
+                ++i;
+            }
+        }
 
-                /// We'd like to compute some reserved column positions for `the left block join right block` case
-                /// since we will need swap them before project the join results
-                Block left_join_right_header = left_data.join_stream_desc->input_header;
-                doJoinBlockWithHashTable<true>(left_join_right_header, right_data.buffered_data->getCurrentHashBlocksPtr());
-                for (size_t i = 0; const auto & col : left_join_right_header)
-                {
-                    if (col.name == ProtonConsts::RESERVED_DELTA_FLAG)
-                        left_delta_column_position_lrj = i;
-                    else if (col.name.ends_with(ProtonConsts::RESERVED_DELTA_FLAG))
-                        right_delta_column_position_lrj = i;
+        /// 2) We'd like to compute some reserved column positions for `the left block join right block` case
+        /// since we will need swap them before project the join results
+        Block left_join_right_header = left_data.join_stream_desc->input_header;
+        doJoinBlockWithHashTable<true>(left_join_right_header, right_data.buffered_data->getCurrentHashBlocksPtr());
+        for (size_t i = 0; const auto & col : left_join_right_header)
+        {
+            if (col.name == ProtonConsts::RESERVED_DELTA_FLAG)
+                left_delta_column_position_lrj = i;
+            else if (col.name.ends_with(ProtonConsts::RESERVED_DELTA_FLAG))
+                right_delta_column_position_lrj = i;
 
-                    ++i;
-                }
-            }
+            ++i;
         }
     }
 }
 
 void HashJoin::transformHeader(Block & header)
 {
     if (range_bidirectional_hash_join)
-    {
         joinBlockWithHashTable<true>(header, right_data.buffered_data->getCurrentHashBlocksPtr());
-    }
     else if (bidirectional_hash_join)
-    {
         joinBlockWithHashTable<true>(header, right_data.buffered_data->getCurrentHashBlocksPtr());
+    else
+        joinLeftBlock(header);
 
-        /// Fix the emit changelog header, remove left/right delta, then append joined delta
-        if (emitChangeLog())
+    /// Remove internal left/right delta column
```

Comment:
Do we need guard this with `bidirectional_hash_join` since not all cases have delta column ? we will waste computation if we do this for every join case

Reply:
for `append join changelog` enrichment join, the joined result has `right_kv._tp_delta`, we still need remove it